src/mlpack/core/tree/rectangle_tree/no_auxiliary_information.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file no_auxiliary_information.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Definition of the NoAuxiliaryInformation class, a class that provides
+ * no additional information about the nodes.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_NO_AUXILIARY_INFORMATION_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_NO_AUXILIARY_INFORMATION_HPP
+
+namespace mlpack {
+namespace tree {
+
+template<typename TreeType>
+class NoAuxiliaryInformation
+{
+ public:
+  NoAuxiliaryInformation() { };
+  NoAuxiliaryInformation(const TreeType* ) { };
+  NoAuxiliaryInformation(const TreeType& ) { };
+
+  /**
+   * Some tree types require to save some properties at the insertion process.
+   * This method should return false if it does not handle the process.
+   */
```

Comment:
I spent a while reading through `NoAuxiliaryInformation` and `HilbertRTreeAuxiliaryInformation` and it took me a long time to understand exactly what `HandlePointInsertion`, `HandleNodeInsertion`, `HandlePointDeletion`, and `HandleNodeRemoval` did (like what the parameters were, what the return value should be).

To me it seems like each of these methods allow the auxiliary information the option of manipulating the tree to perform the insertion/deletion itself, but if the auxiliary information does that, then it should return true; if false is returned, the RectangleTree performs its default behavior.

Do you think you could add some documentation to both this file and HilbertRTreeAuxiliaryInformation detailing this?  (and what the HilbertRTreeAuxiliaryInformation does?)  That way someone else who wants to add some different auxiliary information class in the future (or just wants to understand these) will have a much easier time.